internal/subscriber/subscriber.go

```diff
@@ -205,6 +216,11 @@ func (s *Subscriber) processUpdateEvent(ctx context.Context, event tmtypes.Resul
 				zap.String("query_id", queryID))
 			continue
 		}
+		if !s.isWatchedQueryID(neutronQuery.Id) {
+			s.logger.Debug("Skipping query (wrong ID)", zap.String("owner", owner),
+				zap.String("query_id", queryID))
+			continue
+		}
```

Comment:
query ID is already known [above](https://github.com/stafiprotocol/neutron-query-relayer/blob/26395bb01d3b6ae6fba52a917734a8077fbab9ad/internal/subscriber/subscriber.go#L200) before the [query information loading](https://github.com/stafiprotocol/neutron-query-relayer/blob/26395bb01d3b6ae6fba52a917734a8077fbab9ad/internal/subscriber/subscriber.go#L208-L212). it's better to move this check before the loading to improve efficiency by removing the loading call to the node